concepts/0094-cross-domain-messaging/README.md

```diff
@@ -0,0 +1,302 @@
+  RFCs about routing. Supersedes Indy RFC [0022-Cross-Domain-Messaging](https://github.com/hyperledger/indy-RFC/blob/master/text/0022-cross-domain-messaging/README.md)
+
+## Summary
+
+There are two layers of messages that combine to enable **interoperable** self-sovereign identity DIDcomm (formerly called Agent-to-Agent) communication. At the highest level are Agent Messages - messages sent between Identities to accomplish some shared goal. For example, establishing a connection between identities, issuing a Verifiable Credential from an Issuer to a Holder or even the simple delivery of a text Instant Message from one person to another. Agent Messages are delivered via the second, lower layer of messaging - encryption envelopes. An encryption envelope is a wrapper (envelope) around an Agent Message to enable the secure delivery of a message from one Agent directly to another Agent. An Agent Message going from its Sender to its Receiver may be passed through a number of Agents, and an encryption envelope is used for each hop of the journey.
+
+This RFC addresses Cross Domain messaging to enable interoperability. This is one of a series of related RFCs that address interoperability, including [DIDDoc Conventions](../../features/0067-didcomm-diddoc-conventions/README.md), [Agent Messages](../0020-message-types/README.md) and [Encryption Envelope](../../features/0019-encryption-envelope/README.md). Those RFCs should be considered together in understanding [DIDcomm](../0005-didcomm/README.md) messaging.
+
+In order to send a message from one Identity to another, the sending Identity must know something about the Receiver's domain - the Receiver's configuration of Agents. This RFC outlines how a domain MUST present itself to enable the Sender to know enough to be able to send a message to an Agent in the domain.  In support of that, a DIDcomm protocol (currently consisting of just one Message Type) is introduced to route messages through a network of Agents in both the Sender and Receiver's domain. This RFC provides the specification of the "Forward" Agent Message Type - an envelope that indicates the destination of a message without revealing anything about the message.
+
+The goal of this RFC is to define the rules that domains MUST follow to enable the delivery of Agent messages from a Sending Agent to a Receiver Agent in a secure and privacy-preserving manner.
+
+## Motivation
+
+The purpose of this RFC and its related RFCs is to define a layered messaging protocol such that we can ignore the delivery of messages as we discuss the much richer Agent Messaging types and interactions. That is, we can assume that there is no need to include in an Agent message anything about how to route the message to the Receiver - it just magically happens. Alice (via her App Agent) sends a message to Bob, and (because of implementations based on this series of RFCs) we can ignore how the actual message got to Bob's App Agent.
```

Comment:
@swcurran nitpick: I think "we can ignore the **transport** of messages" was a more accurate term, unless I'm missing some context?

Reply:
I changed that right at the end from what was in the previous.  I was trying to get away from the notion that it was involved in the transport layer protocols - e.g. http, etc. That's a totally separate issue from this one.  As is stated elsewhere, DIDcomm is agnostic to transport and should work using a variety of protocols, including smoke signals, if carefully generated and equally carefully recorded and converted to bytes.